### ISSUE TITLE
[Other - Accessibility] Destructive action with no confirmation step or screen reader feedback. (00.00.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#56082]
#### VFS team name
Benefits Team 1

#### VFS product name
Supplemental Claims Form 20-0995

#### VFS feature name
Supplemental Claims Form 20-0995

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
No VA.gov Experience Standard for the issue found.

### Issue Details
On [Step 2 of 4: Issues for review - Select the issues you'd like us to review, when a user adds a new issue and [Step 3 of 4: New and relevant evidence - Confirm or edit your evidence], the Remove button is a single-click destructive action. When clicking Remove, the item is removed without any user confirmation. Additionally, there's nothing announced by screen readers to indicate that a change has occurred --- focus just moves to the next focusable element.

It's easy to imagine some scenarios where a user might accidentally trigger the Remove button and not realize that they did it. Screen reader users may hit the wrong key but think they hit [Tab]. A low-vision user on a touch device might pinch-to-zoom and accidentally tap the Remove button in the process, and not see the issue disappear.

Since users get an opportunity to review everything prior to submission, this is probably not a violation of experience standard 03.09 or [WCAG 3.3.4](https://www.w3.org/WAI/WCAG21/Understanding/error-prevention-legal-financial-data.html). But I do think it's in the spirit of WCAG 3.3.4.

A similar issue exists on Step 3 of 4: New and relevant evidence - Upload your supporting evidence, which follows the [design system Files pattern](https://design.va.gov/patterns/ask-users-for/files). I'll be filing an issue with the DST to address this in that pattern as well.

### Success criteria
- [ ] Step 2 of 4: Issues for review
- [x] Step 3 of 4 New and relevant evidence

### VA.gov Experience Standard
[Category Number 00, Issue Number 00](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Platform Recommendation
Two recommended changes:
1. Some kind of status message that receives focus when an issue has been removed, similar to the background-only alerts in use when changing contact information. If a visible status message isn't practical, at least some `aria-live` update should be announced to screen reader users.
2. Some kind of intermediate confirmation step between a user clicking Remove and the issue actually being removed. I hate to recommend a modal for anything ever, but maybe a modal?

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by eileen-coforma** [comment]
[SKETCH mockups](https://www.sketch.com/s/d2416db4-9a4f-4919-abe4-20ba4bdcfd89/v/78AxwP/p/274ABD98-987B-4C4D-BB93-8FCE3DD6C3DE/canvas?posX=-7580.159151539127&posY=-413.1419051462275&zoom=0.9230076670646667)

@Mottie Let me know if you have questions about the modals.

**Comment by eileen-coforma** [comment]
Updated the design for [remove confirmation](https://www.sketch.com/s/d2416db4-9a4f-4919-abe4-20ba4bdcfd89/v/48xllq/p/274ABD98-987B-4C4D-BB93-8FCE3DD6C3DE/canvas?posX=-7958.2708535901875&posY=-425.66399332691833&zoom=0.9230076670646667) and [location confirmation](https://www.sketch.com/s/d2416db4-9a4f-4919-abe4-20ba4bdcfd89/a/ag43VnP). 

The button copy has been shortened. Let me know if the limitation one is still too long. We also need to contract `We will` to `We'll` in the body copy.

**Comment by saderagsdale** [comment]
@joshkimux Hey! Can you check this one out and let us know if it looks good?

**Comment by joshkimux** [comment]
https://user-images.githubusercontent.com/14154792/231861679-3a8599b1-d1fc-4326-940d-4afa062165d9.mov

<br/>

Works like a dream for step 3/4, reviewing evidence


https://user-images.githubusercontent.com/14154792/231861481-ac3d270b-481f-4cf0-952a-388262dff2b1.mov

<br/>

Has not yet been implemented (at least on staging) for issues in step 2/4, was this an intentional choice? @eileen-coforma I don't see I see it in the sketch screenshots.

**Comment by eileen-coforma** [comment]
@joshkimux Good catch. That would be considered a destructive action, and I missed it earlier.

@Mottie Added a [new design](https://www.sketch.com/s/d2416db4-9a4f-4919-abe4-20ba4bdcfd89/a/ellkgEz) to follow the same patterns.

**Comment by joshkimux** [comment]
@saderagsdale will validate this today!

**Comment by joshkimux** [comment]
https://user-images.githubusercontent.com/14154792/232535353-62b0ed1d-96bc-4563-a2cc-365c3be8ff84.mov

This may have not made it to staging yet, will test again later today

**Comment by joshkimux** [comment]
Confirming this has been fixed! Closing out 😄 

https://user-images.githubusercontent.com/14154792/232774056-e87eb8f8-e11e-4064-9d96-d53ab0f8ce7d.mov
